adoc/admin-updates.adoc

```diff
@@ -19,15 +19,19 @@ Updating of {kube} components is handled via `skuba`.
 
 The general procedure should look like this:
 
-. Update the packages and reboot your management workstation to get all the latest changes to skuba and it's dependencies
-. Run `skuba addon upgrade apply` on the management workstation
-.
-. Check if there are addon upgrades available for the current cluster
-. Upgrade all master nodes
-. Upgrade all worker nodes
-. Check if new addons are available for the new version
-.. If necessary upgrade master nodes again
-.. If necessary upgrade worker nodes again
+. Update the packages and reboot your management workstation to get all the latest changes to skuba, helm and their dependencies
+. Run `skuba addon upgrade plan` and then `skuba add upgrade apply` on the management workstation
+. Apply all the configuration files that you modified for addons, the upgrade will have reset the configurations to defaults
+. Check if there are addon upgrades available for the current cluster using `skuba addon upgrade plan`
+.. Check if all the deployments in the cluster are compatible with the {kube} release that will be installed (refer to your individual deployments' documentation)
+If the deployment is not compatible you must update it to ensure it working with the updated {kube}.
+. Upgrade all master nodes by individually running `skuba node upgrade plan` and `skuba node upgrade apply`
+. Upgrade all worker nodes by individually running `skuba node upgrade plan` and `skuba node upgrade apply`
+. Check if new addons are available for the new version using `skuba addon upgrade plan`
+.. If necessary upgrade master nodes and re-apply configuration changes again
+.. If necessary upgrade nodes and re-apply configuration changes again
+. Once all nodes are up to date, update helm and tiller and subsequently the helm deployments
+. If you have other deployments not installed via {kube} or helm, update them last
```

Comment:
Some need to be updated first..

Reply:
So how would you determine which those are and at what point would that check need to happen?

Reply:
Also, can you give a specific example of which ones would have to be updated first?
And by first you mean "before attempting to update the cluster" right?

Reply:
One example would be the k8s dashboard 1.10.1 which does not support kubernetes 1.16. So you might have to update the dashboard i.e. to 2.0.0-rc2 before upgrading kubernetes.

Reply:
> 
> 
> So how would you determine which those are and at what point would that check need to happen?

I am not sure about this but release notes of the applications deployed? Check of the API versions used?